File: 05_cdk_pipelines/python/cdk_pipelines/cdk_pipelines_stack.py
from constructs import Construct
from aws_cdk import (
    Stack,
    aws_codecommit as codecommit,
    pipelines as pipelines,
)
from .pipeline_stage import PipelineStage

class CdkPipelinesStack(Stack):

    def __init__(self, scope: Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        repo = codecommit.Repository(
            self, 'cdkPipelinesRepo',
            repository_name= "cdkPipelinesRepo"
        )

        # Pipeline code goes here
        pipeline = pipelines.CodePipeline(
            self,
            "cdkPipelines",
            synth=pipelines.ShellStep(
                "Synth",
                input=pipelines.CodePipelineSource.code_commit(repo, "master"),
                commands=[
                    "npm install -g aws-cdk",  # Installs the cdk cli on Codebuild
                    "pip install -r requirements.txt",  # Instructs Codebuild to install required packages
                    "npx cdk synth",
                ]
            ),
        )
        deploy = PipelineStage(self, "Deploy")
        deploy_stage = pipeline.add_stage(deploy)

File: 05_cdk_pipelines/typescript/lib/cdk_pipelines-stack.ts
import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as codecommit from 'aws-cdk-lib/aws-codecommit';
import {CodeBuildStep, CodePipeline, CodePipelineSource} from "aws-cdk-lib/pipelines";
import {PipelineStage} from './pipeline-stage';

export class CdkPipelinesStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const repo = new codecommit.Repository(this, 'cdkPipelinesRepo', {
      repositoryName: "cdkPipelinesRepo"
    });

    const pipeline = new CodePipeline(this, 'Pipeline', {
      pipelineName: 'WorkshopPipeline',
      synth: new CodeBuildStep('SynthStep', {
              input: CodePipelineSource.codeCommit(repo, 'master'),
              installCommands: [
                  'npm install -g aws-cdk'
              ],
              commands: [
                  'npm ci',
                  'npm run build',
                  'npx cdk synth'
              ]
          }
      )
    });

    const deploy = new PipelineStage(this, 'Deploy');
    const deployStage = pipeline.addStage(deploy);
  }
}
